### DISCUSSION TITLE
visual regression testing

### DISCUSSION THREAD

**Discussion by Eduruiz** [getgauge/taiko#2526] [Ideas]
Should taiko incorporate a visual regression test option? I found [a plugin](https://github.com/dcdc-io/taiko-screencheck) to do it but couldn't make it work, but the discussion here is more about this being incorporated as a function of the main taiko repo.
I'm implementing and loved the simplicity of writing things with taiko, but since I want visual regression too (and primary to be honest) I missed this feature and this made me choose [playwrite](playwright) at the moment.
Because taiko runs locally I imagine that building this feature is not simple if we think about cross-browser compatibility, for example, playwrite suffer from the same problem with lots of dependencies to simulate safari in linux for example
But it would be awesome to see something like this working on taiko :smiley:

**Comment by zabil** [comment]
Hey @Eduruiz thanks bringing up this discussion. Unfortunately, there are no plans to add this feature to Taiko. The ideal way is to implement it using a custom plugin for now. 
But again this would only work on chrome as Taiko uses chrome devtools protocol which is not fully implemented on other browsers.

**Reply by Eduruiz** [reply]
Got it! Sad to hear but understandable since the focus of the project is not in this direction and the complexity of the implementation seems really high.
Thanks for the reply and for this awesome tool, looking forward to using it in a project with another requirements :)
